src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -282,171 +315,164 @@ export default class OfficeAddresses extends Mixins(CommonMixin, EntityFilterMix
-        this.recDeliveryAddress = this.addresses.recordsOffice?.deliveryAddress
-
-        if (loadInheritedFlags) {
-          this.inheritRegisteredAddress = (
-            this.isSame(
-              this.addresses.registeredOffice.deliveryAddress,
-              this.addresses.recordsOffice?.deliveryAddress
-            ) && this.isSame(
-              this.addresses.registeredOffice.mailingAddress,
-              this.addresses.recordsOffice?.mailingAddress
-            )
-          )
+  /** True if records delivery address has changed. */
+  private get recDeliveryChanged (): boolean {
+    return !this.isSame(this.getOfficeAddresses.recordsOffice?.deliveryAddress,
+      this.originalOfficeAddresses.recordsOffice?.deliveryAddress)
+  }
 
-          this.inheritRecMailingAddress = this.isSame(
-            this.addresses.recordsOffice?.mailingAddress,
-            this.addresses.recordsOffice?.deliveryAddress
-          )
-        }
-      }
-    }
+  /** True if any office address has changed. */
+  private get officeAddressesChanged (): boolean {
+    return (this.mailingChanged || this.deliveryChanged || this.recMailingChanged || this.recDeliveryChanged)
   }
 
-  //
-  // Getters (Computed Values)
-  //
-  /** Whether the address form is valid. */
+  /** True if the address form is valid. */
   private get formValid (): boolean {
     const registeredOfficeValid = this.mailingAddressValid &&
       (this.deliveryAddressValid || this.inheritMailingAddress)
 
     const recordsOfficeValid = this.inheritRegisteredAddress ||
       (this.recMailingAddressValid && (this.inheritRecMailingAddress || this.recDeliveryAddressValid))
 
-    return registeredOfficeValid && recordsOfficeValid
+    return (registeredOfficeValid && recordsOfficeValid)
   }
 
-  //
-  // Event Handlers
-  //
-  /** Sets the Registered Delivery Address to the Registered Mailing Address. */
+  /**
+   * Sets local address data and "inherited" flags from store.
+   */
+  private setLocalProperties (): void {
+    if (this.getOfficeAddresses.registeredOffice) {
+      this.mailingAddress = { ...this.getOfficeAddresses.registeredOffice.mailingAddress }
+      this.deliveryAddress = { ...this.getOfficeAddresses.registeredOffice.deliveryAddress }
+
+      // compare addresses to set the "inherited mailing" flag
+      this.inheritMailingAddress = this.isSame(
+        this.getOfficeAddresses.registeredOffice.mailingAddress,
+        this.getOfficeAddresses.registeredOffice.deliveryAddress
+      )
+
+      // for BCOMPS, also set the Records Address
+      if (this.entityFilter(EntityTypes.BCOMP)) {
+        this.recMailingAddress = { ...this.getOfficeAddresses.recordsOffice?.mailingAddress }
+        this.recDeliveryAddress = { ...this.getOfficeAddresses.recordsOffice?.deliveryAddress }
+
+        // compare addresses to set the "inherit registered" flag
+        this.inheritRegisteredAddress = (
+          this.isSame(
```

Comment:
this little method is sure used a lot 😁 , i'm sure you know already but it can omit props from the comparison too 😆

Reply:
Yep. I saw that Shares/Series uses it too (also without omitting props ;) ).

If we never omit props, I wonder why we're not using Lodash's isEqual()?

(PS Did you notice that some of Lodash's functions are now part of JS, eg ES6? The built-in ones have the potential to be fast, depending on how the browser interpreter implements them.)